units/schedule_patch.go

```diff
@@ -67,3 +85,34 @@ func SchedulePatch(ctx context.Context, env evergreen.Environment, patchId strin
 	}
 	return http.StatusOK, nil
 }
+
+// addPreviousSingleHostTaskGroupTasks checks all the tasks in a task group in the patch update
+// and if that task group is a single host task group, it schedules the previous tasks in the
+// task group.
+func addPreviousSingleHostTaskGroupTasks(patchUpdateReq *model.PatchUpdate, project *model.Project) {
+	taskGroupTasksToAddToVariant := map[string]string{}
+	for _, vt := range patchUpdateReq.VariantsTasks {
+		for _, t := range vt.Tasks {
+			tg := project.FindTaskGroupForTask(t)
+			if tg == nil || tg.MaxHosts != 1 {
+				continue
+			}
+			for _, tgt := range tg.Tasks {
+				if tgt == t {
+					break
+				}
+				taskGroupTasksToAddToVariant[tgt] = vt.Variant
+			}
+		}
+	}
+	variantTasks := []patch.VariantTasks{}
```

Comment:
I believe `PatchUpdate` has to include all the new variants/tasks that are being scheduled and all previous variants/tasks that have already been scheduled previously by the patch, so just the diff. A similar bug was just fixed in #8051.

Writing a test for appending (rather than overwriting) the variants/tasks would also be good!